src/core/toxid.h

```diff
@@ -41,9 +41,9 @@ class ToxId
 public:
     ToxId(); ///< The default constructor. Creates an empty Tox ID.
     ToxId(const ToxId& other); ///< The copy constructor.
-    ToxId(const QString& id); ///< Create a Tox ID from QString.
-                              /// If the given id is not a valid Tox ID, then:
-                              /// publicKey == id and noSpam == "" == checkSum.
+    explicit ToxId(const QString& id); ///< Create a Tox ID from QString.
+                                       /// If the given id is not a valid Tox ID, then:
+                                       /// publicKey == id and noSpam == "" == checkSum.
```

Comment:
Btw. through the discussion I learned, that Qt also ships the STL and other 3rd party (offline) docs. Didn't know that. You might want to have a look over to https://wiki.qt.io/Qt_Creator_Documentation_Gallery

There's several advantages to keeping the documentation near the function definition:

1. The biggest reason is, that every header can be documented in cpp, while this is not true vice versa. This allows to keep documentation in one place, keeping it more clear.
2. Changes to documentation can never change the header, which in turn doesn't have to be re-shipped.
3. Internal API documentation cannot become public by accident. Even if there's not really an internal documentation existing in qTox, this makes it future proof.
4. One can quickly check, if the function actually does, what the docs say, which comes in handy on a code review.
5. …

@Diadlo If you didn't know already, in Qt Creator you can jump quickly to the function declaration by the F2 or Shift+F2 key. So it's actually easy to find documentation.